src/providers/apns.rs

```diff
@@ -69,33 +67,45 @@ impl PushProvider for ApnsProvider {
         };
 
         // TODO tidy after https://github.com/WalletConnect/a2/issues/67 is closed
-        let result = match payload.is_encrypted() {
-            true => {
-                let mut notification_payload = a2::DefaultNotificationBuilder::new()
-                    .set_content_available()
-                    .set_mutable_content()
-                    .set_title("You have new notifications. Open to view")
-                    .build(token.as_str(), opt);
+        // Sending `always_raw` encrypted message
+        let result = if body.topic.is_some() && body.tag.is_some() && body.message.is_some() {
```

Comment:
It should only do this when `always_raw=true` not when these flags are `Some`

Reply:
Do these fields are required when the `always_raw=true` by the spec? So we know that the client is `always_raw=true` when they are present?

Reply:
Btw, I think [these two](https://github.com/WalletConnect/echo-server/pull/281/files#diff-b90d908d3c5121717714b75753e7f32aec4b723e226dfa3c270e3510fe9d63beR63-R64) should become optional, because of the new client.

Reply:
> Do these fields are required when the always_raw=true by the spec?

Yes, you can't support clients with `always_true` if the relay hasn't been upgraded yet to support. In practice this usually won't happen except during initial deployment, and we most likely will deploy the relay update first anyway.

> So we know that the client is always_raw=true when they are present?

No, the relay has no knowledge of `always_true` value, and relay upgraded with [this PR](https://github.com/WalletConnect/rs-relay/pull/1285) will always send the fields.

> Btw, I think [these two](https://github.com/WalletConnect/echo-server/pull/281/files#diff-b90d908d3c5121717714b75753e7f32aec4b723e226dfa3c270e3510fe9d63beR63-R64) should become optional, because of the new client.

Not a bad idea and would enable smoother migration later if we remove these flags from the relay, so makes sense. If `always_true=true` then require the new fields, and if `always_true=false` then require the old fields. Could use `#[serde(flatten)]` potentially to avoid checking more than 2 `Option`s

Reply:
Thanks for clarifying! 

> Btw, I think [these two](https://github.com/WalletConnect/echo-server/pull/281/files#diff-b90d908d3c5121717714b75753e7f32aec4b723e226dfa3c270e3510fe9d63beR63-R64) should become optional, because of the new client.

> Not a bad idea and would enable smoother migration later if we remove these flags from the relay, so makes sense. If always_true=true then require the new fields, and if always_true=false then require the old fields. Could use #[serde(flatten)] potentially to avoid checking more than 2 Options

I think making fields optional should be a follow-up.

Reply:
> It should only do this when always_raw=true not when these flags are Some

Fixed. [Check for required fields](https://github.com/WalletConnect/echo-server/pull/281/files#diff-b90d908d3c5121717714b75753e7f32aec4b723e226dfa3c270e3510fe9d63beR196-R206) is in the handler and [in the test](https://github.com/WalletConnect/echo-server/pull/281/files#diff-513a9ee6d914ac38439771435145221152a2aeafff7c1a41aba304842354ec79R207-R226) now.